libethash/io_posix.c

```diff
@@ -86,9 +90,22 @@ bool ethash_file_size(FILE* f, size_t* ret_size)
 
 bool ethash_get_default_dirname(char* strbuf, size_t buffsize)
 {
+#if defined(__APPLE__)
+	static const char dir_suffix[] = "Ethash/";
+	strbuf[0] = '\0';
+	char* home_dir = getenv("HOME");
+	if (!home_dir || strlen(home_dir) == 0)
```

Comment:
If you need the `getpwuid()` call for Mac put it in but remove the ifndef. It will work on Linux too.

As for the "Application Data" Folder, Windows isn't Posix compliant but Macosx ... mostly ... is. Why change it to a specific application data folder for Mac? Doesn't Mac have the notion of a user's home directory? Can't an application create a subdirectory in there called `.ethash`? If yes then why bother having it in a different location?

Windows has and always will be an exception. Macosx should be only when it's absolutely necessary. Of course not being a MacOS user I may be missing something so please bear with my questions.